storage/src/block_meta_storage.rs

```diff
@@ -565,7 +547,7 @@ fn merge_meta_value(
                 }
 
                 let total_len = total_len(op_successors_count);
-                assert_eq!(total_len, val.len(), "Invalid length after merge operator was applied. Was expecting {} but found {}.", total_len, val.len());
+                debug_assert_eq!(total_len, val.len(), "Invalid length after merge operator was applied. Was expecting {} but found {}.", total_len, val.len());
```

Comment:
This looks like a check for correct calculation, so `debug_assert` looks appropriate.